cmd/daemon/kubernetes/kubernetes.go

```diff
@@ -1,332 +1,58 @@
 package kubernetes
 
 import (
-	"bytes"
-	"context"
-	"encoding/json"
-	"fmt"
-	"io"
-	"strings"
-
-	"github.com/lunarway/release-manager/internal/log"
 	"github.com/pkg/errors"
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"time"
+
+	//deploymentutil "k8s.io/kubectl/pkg/util/deployment"
+
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
 type Client struct {
-	clientset *kubernetes.Clientset
-	config    *rest.Config
+	clientset              *kubernetes.Clientset
+	exporter               Exporter
+	moduloCrashReportNotif float64
+	replicaSetTimeDiff     time.Duration
 }
 
 var (
 	ErrWatcherClosed = errors.New("channel closed")
 )
 
-func NewClient(kubeConfigPath string) (*Client, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
-	if err != nil {
-		return nil, err
-	}
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-
-	client := Client{
-		clientset: clientset,
-		config:    config,
-	}
-
-	return &client, nil
-}
-
-func (c *Client) GetLogs(podName, namespace string) (string, error) {
-	numberOfLogLines := int64(8)
-	podLogOpts := v1.PodLogOptions{
-		TailLines: &numberOfLogLines,
-	}
-	req := c.clientset.CoreV1().Pods(namespace).GetLogs(podName, &podLogOpts)
-
-	podLogs, err := req.Stream()
-	if err != nil {
-		return "", err
+func NewClient(kubeConfigPath string, moduloCrashReportNotif float64, replicaSetTimeDiff time.Duration, e Exporter) (*Client, error) {
+	if kubeConfigPath != "" {
```

Comment:
There is no need to handle this case manually. `BuildConfigFromFlags` will fallback to `InClusterConfig` if the path is empty.
https://godoc.org/k8s.io/client-go/tools/clientcmd#BuildConfigFromFlags

Reply:
Ah, I didn't know. That's awesome 🎉 So much easier!